yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -82,6 +100,179 @@ class YarnSparkHadoopUtil extends SparkHadoopUtil {
     if (credentials != null) credentials.getSecretKey(new Text(key)) else null
   }
 
+  private[spark] override def scheduleLoginFromKeytab(): Unit = {
+    sparkConf.getOption("spark.yarn.principal").foreach { principal =>
+      val keytab = sparkConf.get("spark.yarn.keytab")
+      val delegationTokenRenewerRunnable =
```

Comment:
The reason it is not stashed away in a lazy is because the principal and keytab would need to be instance vars if we did. By doing this, I can avoid that.